staging/src/k8s.io/api/core/v1/types.go

```diff
@@ -450,6 +450,13 @@ type PersistentVolumeClaimSpec struct {
 	// More info: https://kubernetes.io/docs/concepts/storage/persistent-volumes#resources
 	// +optional
 	Resources ResourceRequirements `json:"resources,omitempty" protobuf:"bytes,2,opt,name=resources"`
+	// AllocatedResources represents the resources that was requested to satisfy user
+	// request. This field is not meant to be set by the user but is used for tracking
+	// volume quota.
+	// This field is alpha-level and is only honored by servers that enable the RecoverVolumeExpansionFailure feature.
+	//
+	// +optional
+	AllocatedResources *ResourceRequirements `json:"allocatedResources,omitempty" protobuf:"bytes,8,opt,name=allocatedResources"`
```

Comment:
If I've understood the usage, this a value on spec which is derived on updates to spec to track "max value ever requested".  A user could request a different volume size, but it wouldn't impact his quota and I'm not sure it actually impacts the reality of the size allocated for him.

One thing I don't understand is that why this information would not be status information provided when the expansion actually happened.

Reply:
Hmm.. I am not opposed to move this field to Status. See discussion in KEP - https://github.com/kubernetes/enhancements/pull/1516#discussion_r427017345

This is something I discussed with @liggitt at kubecon and I used in-place pod resizing KEP as a inspiration/starting point to store `allocatedResources` - https://github.com/kubernetes/enhancements/pull/1342/files